packages/create-react-admin/src/cli.tsx

```diff
@@ -6,31 +6,88 @@ import App from './app.js';
+    }
+    return dataProvider;
+};
+
+const getDataProvider = (flags: typeof cli.flags) => {
+    if (flags.dataProvider) return getDataProviderName(flags.dataProvider);
+    if (flags.interactive) return undefined;
+    return 'none';
+};
+
+const authProviderShortcuts = {
+    local: 'local-auth-provider',
+};
+
+const getAuthProviderName = (authProvider: string | undefined) => {
+    if (authProviderShortcuts[authProvider]) {
+        return authProviderShortcuts[authProvider];
+    }
+    return authProvider;
+};
+
+const getAuthProvider = (flags: typeof cli.flags) => {
+    if (
+        getDataProviderName(flags.dataProvider) === 'ra-supabase' &&
+        flags.authProvider != null
+    ) {
+        console.error("Don't provide an auth-provider when using ra-supabase");
+        process.exit(1);
+    }
+    if (flags.authProvider) return getAuthProviderName(flags.authProvider);
+    if (flags.dataProvider === 'ra-supabase') return 'none';
+    if (flags.interactive) return undefined;
+    return 'none';
+};
+
+const getInstall = (flags: typeof cli.flags) => {
+    if (flags.install) return flags.install;
+    if (flags.interactive) return undefined;
+    return 'npm';
+};
+
+const getResources = (flags: typeof cli.flags) => {
+    if (flags.resource.length > 0) return flags.resource;
+    if (flags.interactive) return undefined;
+    return ['skip'];
+};
+
 const cli = meow(
     `
 	Usage
 	  $ create-admin-app <name>
 
     Options
-	  --data-provider  Set the data provider to use ("ra-data-fakerest", "ra-data-simple-rest", "ra-data-json-server" or "none")
+	  --interactive    Enable the CLI interactive mode
+	  --data-provider  Set the data provider to use ("ra-data-fakerest", "ra-data-simple-rest", "ra-data-json-server", "ra-supabase" or "none")
 	  --auth-provider  Set the auth provider to use ("local-auth-provider" or "none")
 	  --resource       Add a resource that will be initialized with guessers (can be used multiple times). Set to "skip" to bypass the interactive resource step.
-	  --install        Set the package manager to use for installing dependencies ("yarn", "npm" or "skip" to bypass the interactive install step)
-	  --basic          Skip all the interactive steps and create a basic app with no data provider, no auth provider, no resources and install with npm
+	  --install        Set the package manager to use for installing dependencies ("yarn", "npm", "pnpm", "bun" or "skip" to bypass the interactive install step)
 
     Examples
 	  $ create-admin-app my-admin
 	  $ create-admin-app my-admin --data-provider ra-data-json-server --auth-provider local-auth-provider --resource posts --resource comments --install npm
-	  $ create-admin-app my-admin --basic
 `,
     {
         flags: {
             help: {
                 type: 'boolean',
                 alias: 'h',
             },
-            basic: {
+            interactive: {
                 type: 'boolean',
+                alias: 'i',
```

Comment:
I see you did not update the choices for the `install` flag a few lines below (it's still `choices: ['yarn', 'npm', 'skip']`). Aren't the possible choices checked? :thinking: Because I'm still able to pick 'bun' or 'pnpm' according to my tests, so I'm surprised it works...

Reply:
It looks like they're not checked, I fixed them anyway